pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -600,6 +609,29 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileClusterDeployment) reconcileLogsPVC(cd *hivev1.ClusterDeployment, pvc *corev1.PersistentVolumeClaim, cdLog log.FieldLogger) error {
+	if err := controllerutil.SetControllerReference(cd, pvc, r.scheme); err != nil {
```

Comment:
Move this down to the section where the PVC is created. There is no need to do this except in that case.